src/components/profile/UserInfo.js

```diff
@@ -1,18 +1,51 @@
 import React from "react";
 import SelectedListItem from "../ui/SelectedListItem";
-
-const user = {
-  firstName: "John",
-  middleName: "A.",
-  lastName: "Doe",
-  gender: "Male",
-  dateOfBirth: "1990-05-15",
-  phone: "1234567890",
-  email: "john.doe@example.com",
-  city: "New York",
-  country: "USA",
-};
+import { useEffect, useState } from "react";
+import axios from "axios";
 
 export default function UserInfo() {
-  return <SelectedListItem list={user} />;
+  const token = localStorage.getItem("token");
+  const [userData, setUserData] = useState(null);
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      if (!token) {
+        console.error("No token found. Please log in.");
```

Comment:
You shouldn't use console, instead using toast.

Reply:
fixed